OpenAPI: errors under nodejs 10

When executing under Nodejs 10 (the minimum currently supported LTS version), the backend test (`cd src ; npm test`) cause an error in the server log.

Etherpad version: e6251687bf6f (current develop).
Tested under:
* linux, nodejs **12.13.1** <-- works
* linux, nodejs **10.14.1** <-- error

This is the relevant snippet:
```
[2020-04-05 02:50:47.813] [INFO] API - REQUEST, v1.2.14:setHTML, {"apikey":"XXXX","padID":"pWvFw","html":"<div><b>Hello HTML</title></head></div>"}
[2020-04-05 02:50:47.815] [WARN] ImportHtml - HTML was not properly formed TypeError: Cannot read property 'tagName' of undefined
    at Object.nodeTagName (<BASE>/src/static/js/contentcollector.js:50:16)
    at isBlockElement (<BASE>/src/static/js/contentcollector.js:97:31)
    at Object.cc.collectContent (<BASE>/src/static/js/contentcollector.js:395:19)
    at Object.exports.setPadHTML (<BASE>/src/node/utils/ImportHtml.js:41:8)
    at Object.exports.setHTML (<BASE>/src/node/db/API.js:290:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)
TypeError: Cannot read property 'tagName' of undefined
    at Object.nodeTagName (<BASE>/src/static/js/contentcollector.js:50:16)
    at isBlockElement (<BASE>/src/static/js/contentcollector.js:97:31)
    at Object.cc.collectContent (<BASE>/src/static/js/contentcollector.js:395:19)
    at Object.exports.setPadHTML (<BASE>/src/node/utils/ImportHtml.js:41:8)
    at Object.exports.setHTML (<BASE>/src/node/db/API.js:290:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)
[2020-04-05 02:50:47.818] [ERROR] console - (node:52817) [DEP0079] DeprecationWarning: Custom inspection function on Objects via .inspect() is deprecated
```

I noticed this when trying to run jshint, and it complained that something was only supported on ECMAScript 9 (I had to enable `esversion: 9` in `.jshintrc`).

More details later.

@anttiviljami: can you please investigate? Thanks


## Etherpad NodeJS support:
Etherpad **1.8.0** supports node 8 (see the startup logs) and the documentation.
Etherpad **1.8.3** will require node 10 (and is required to support it). See #3650. The next version will be called 1.8.3 (bad milestone name)

So the test that triggers this error is this one:

https://github.com/ether/etherpad-lite/blob/e6251687bf6f0d773ef5cd8883465d67e18f4816/tests/backend/specs/api/pad.js#L544-L558

It deliberately tries to set incorrect HTML, so the warning message is actually correct behaviour.

```
[WARN] ImportHtml - HTML was not properly formed TypeError: Cannot read property 'tagName' of undefined
```

With Node 10 however it is also accompanied with an error-level deprecation message about using custom `.inspect()` methods.
```
[ERROR] console - (node:52817) [DEP0079] DeprecationWarning: Custom inspection function on Objects via .inspect() is deprecated
```

Investigating further, this seems to happen when trying to inspect an error generated by the customError module found here: https://github.com/ether/etherpad-lite/blob/e6251687bf6f0d773ef5cd8883465d67e18f4816/src/node/utils/customError.js

Rewriting the customError module using modern class syntax fixes the issue for node 10 and above.

--- a/src/node/utils/customError.js
+++ b/src/node/utils/customError.js
@@ -1,17 +1,23 @@
-/*
- This helper modules allows us to create different type of errors we can throw
-*/
-function customError(message, errorName)
-{
-  this.name = errorName || "Error";
-  this.message = message;
-
-  var stackParts = new Error().stack.split("\n");
-  stackParts.splice(0,2);
-  stackParts.unshift(this.name + ": " + message);
-
-  this.stack = stackParts.join("\n");
+/**
+ * CustomError
+ *
+ * This helper modules allows us to create different type of errors we can throw
+ *
+ * @class CustomError
+ * @extends {Error}
+ */
+class CustomError extends Error {
+  /**
+   * Creates an instance of CustomError.
+   * @param {*} message
+   * @param {string} [name='Error'] a custom name for the error object
+   * @memberof CustomError
+   */
+  constructor(message, name = 'Error') {
+    super(message);
+    this.name = name;
+    Error.captureStackTrace(this, this.constructor);
+  }
 }
-customError.prototype = Error.prototype;
 
-module.exports = customError;
+module.exports = CustomError;

